src/Zenaton/v1/Runtime/Contexts/TaskContext.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Zenaton\Runtime\Contexts;
+
+/**
+ * Represents the current runtime context of a class.
+ *
+ * The information provided by the context can be useful to alter the behaviour of the task.
+ * For example, you can use the attempt index to know if a task has been automatically retried or not and how many times,
+ * and decide to do something when you did not expect the task to be retried more than X times.
+ * You can also use the attempt number in the `onErrorRetryDelay` method of a task in order to implement complex
+ * retry strategies.
+ */
+final class TaskContext
+{
+    /**
+     * The current attempt index.
+     *
+     * When the task is executed for the first time, the value will be `1`.
+     * Every time the task is automatically retried, the value is increased.
+     * When a manual retry is instructed, the value goes back to `1`.
+     *
+     * @var null|int
+     */
+    public $attemptIndex;
```

Comment:
I think this more of a counter than an index. Since it gets only incremented when doing automatic retries, I'd suggest naming it `automaticAttemptCount`.

What do you think?

Reply:
We started to discuss that with @geomagilles at some point. We had `attemptNumber` before which sound weird to me.

The count thing can be  misleading to me because I'm not expecting it to go back to 1 but this one does. When you manually retry, it goes back to one. So you total counter of attempt is 12 but your automaticAttemptCount would be 1 because you just manually retried. It's weird.

That's why we went for index for now.